File: skellig-test-step-processing/src/main/java/org/skellig/teststep/processing/validation/TestStepResultValidator.java
package org.skellig.teststep.processing.validation;

import org.skellig.teststep.processing.model.ExpectedResult;
import org.skellig.teststep.reader.exception.ValidationException;

public interface TestStepResultValidator {

    void validate(ExpectedResult expectedResult, Object actualResult) throws ValidationException;
}

File: skellig-test-step-processing/src/main/kotlin/org/skellig/teststep/processing/validation/TestStepResultValidator.kt
package org.skellig.teststep.processing.validation

import org.skellig.teststep.processing.exception.ValidationException
import org.skellig.teststep.processing.model.ExpectedResult

/**
 * Validate actual and expected results.
 */
interface TestStepResultValidator {

    @Throws(ValidationException::class)
    fun validate(expectedResult: ExpectedResult, actualResult: Any?)
}